lib/registries/addon/components/registries-license-picker/template.hbs

```diff
@@ -1,91 +1,89 @@
-        >
-            {{t 'app_components.license_picker.faq'}}
-        </OsfLink>
-    </div>
-
-    {{#if @manager.selectedLicense}}
-        {{#if @manager.requiredFields}}
-            <br>
+    <div data-test-license-edit-form>
+        {{#if @manager.licensesAcceptable}}
+            <form.select
+                data-analytics-category='text'
+                data-analytics-name='Select license'
+                data-test-select-license
+                @id='license-select'
+                @valuePath='license'
+                @selected={{@manager.selectedLicense}}
+                @options={{@manager.licensesAcceptable}}
+                @onchange={{@manager.changeLicense}}
+                @searchField={{'name'}}
+                @noMatchesMessage={{t 'registries.registration_metadata.no_matches'}}
+                @placeholder={{t 'registries.registration_metadata.add_license'}}
+                as |license|
+            >
+                {{license.name}}
+            </form.select>
+        {{else}}
+            <LoadingIndicator @dark={{true}} />
         {{/if}}
-
-        <form.custom @valuePath='nodeLicense'>
-            {{#each @manager.requiredFields as |key|}}
-                <div class='form-group'>
-                    <label>
-                        {{t (concat 'app_components.license_picker.fields.' key)}}
-                        <Input
-                            data-test-required-field={{key}}
-                            @class='form-control'
-                            @value={{mut (get @manager.registration.nodeLicense key)}}
-                            @placeholder={{t 'general.required'}}
-                            @key-up={{action this.onInput}}
-                        />
-                    </label>
+        
+        <div local-class='small help-link'>
+            <OsfLink
+                @target='_blank'
+                @rel='noopener'
+                @href={{this.helpLink}}>
+                {{t 'app_components.license_picker.faq'}}
+            </OsfLink>
+        </div>
+        
+        {{#if @manager.selectedLicense}}
+            {{#if @manager.requiredFields}}
+                <br>
+            {{/if}}
+            <form.custom
+                @changeset={{@manager.changeset}}
+                @valuePath='nodeLicense'
+            >
+                {{#each @manager.requiredFields as |key|}}
+                    <div class='form-group'>
+                        <label>
+                            {{t (concat 'app_components.license_picker.fields.' key)}}
+                            <Input
+                                data-test-required-field={{key}}
+                                @class='form-control'
+                                @value={{readonly (get @manager.changeset.nodeLicense key)}}
+                                @placeholder={{t 'general.required'}}
+                                @change={{action @manager.updateNodeLicense key}}
```

Comment:
oh, one more thought:
```suggestion
                                @change={{fn @manager.updateNodeLicense key}}
```
now that we've upgraded and the `@action` decorator autobinds, there's no need to use the `{{action}}` helper at all -- we can pass action functions around as-is and use `{{fn}}` to partially apply args